workshop/posting-error-example.md

```diff
@@ -0,0 +1,201 @@
+    - [What is the goal of this code?](#what-is-the-goal-of-this-code)
+    - [What is the code that is producing this error?](#what-is-the-code-that-is-producing-this-error)
+    - [What is the error or problematic outcome?](#what-is-the-error-or-problematic-outcome)
+    - [(Optional) What have you tried thus far?](#optional-what-have-you-tried-thus-far)
+  - [Step 3) Post to Slack!](#step-3-post-to-slack)
+  - [Step 4) Look for responses (in the same thread)!](#step-4-look-for-responses-in-the-same-thread)
+  - [Posting a code question checklist](#posting-a-code-question-checklist)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# What information should a question about code include?
+
+We consider the themes of a good troubleshooting post to be `context` and `specificity`.
+The more context you can provide, and the more specific you can be about your problem, the easier it will be for someone to help you.
+
+**Overall your posted code question should include:**
+
+- Where is this error occurring (notebook/chunk)?
+- What is the goal of this code?
+- What is the code that is producing this error?
+- What is the error or problematic outcome?
+
+*Optionally:*
+- What things have you tried thus far?
+
+To demonstrate what we mean by these guidelines we will walk through an example:
+
+# A troubleshooting example
+
+In this example, I'm using our exercise notebook, `04a-intro_to_R_exercise.Rmd`.
+
+Below I'm showing a screenshot of what this error would look like when encountering it in [our RStudio Server](../virtual-setup/rstudio-login.md)
+
+<img src = "screenshots/file.path-error.png" width = "800">
+
+## Step 1) Do what you can to try to solve the error yourself (but don't tire yourself out!)
+
+The best way to learn how to solve errors in code is to figure them out yourself.
+So try to explore the error a bit - but we are also here to help you so don't hesitate to ask for help! (_If you are feeling exasperated by your error, skip to Step 2_).
+
+We recommend looking over our [debugging guide](https://github.com/AlexsLemonade/training-modules/blob/master/intro-to-R-tidyverse/00b-debugging_resources.md) which can explain what some of the most common errors mean.
+The debugging guide also has tips on first steps to take which may help you get to the root of the problem.
+
+**A short list of things to try:**  
+
+- Try to identify which part of your code appears to be the problem through trying smaller parts of the code and seeing if the error still occurs (called chunking).
+In the chunking example below, each highlighted portion is something you could run by itself to test.
+(Using `Command + Enter` in RStudio will run what you have highlighted.)
+
+<img src = "screenshots/chunking_code.png" width = "500">
+
+- Go back to the beginning of your notebook and re-run your code in order and make sure you haven't missed any vital steps! Order matters!
```

Comment:
Do we want to tell them about the down arrow with a line (picture will help here) next to the play button? This will run everything from the top down to the current chunk. Very handy. 

Also, clean out environment first? This should rarely be a problem, but it can be. If we are rerunning everything, might as well.